src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -52,6 +52,7 @@ public async Task<ExecutionResult> ExecuteAsync(ExecutionOptions options)
 
             options.Schema.NameConverter = options.NameConverter;
             options.Schema.Filter = options.SchemaFilter;
+            options.Schema.Comparer = options.SchemaComparer;
```

Comment:
This code does not belong here or in `ExecutionOptions`, but rather should be set by the Schema constructor.  See #1589 .

Reply:
I remember that well. This change is made on purpose to maintain the current style. In version 4, we can change everything.